### PR TITLE
Sets MAT_CATEGORY_ITEM_MATERIAL to TRUE for Bluespace Crystals

### PR DESCRIPTION
**Pull request by Zenitheevee**
## About The Pull Request

<details>
  <summary>Ramblings of a madman</summary>
Tell me, Have you ever wondered what would happen in a perfect world? A perfect world where we can allow Bluespace Crystals inside our autolathes, Where a man can reasonably ask for a print of a bluespace cell from a cargo member who managed to sequester a design disk of it from their fellow cohort, Where we dont have to be wanting for Bluespace Crystals to fit into the autolathe despite there being some hints of code that implies that it is capable of receiving them but for some reason when you go to test it ingame it tells you that it doesn't have enough materials to be allowed in but for some reason we _still_ have that one little bit of code laughing at us, Daring us to try and put the bluespace crystal in with _nothing_ to gain but anger and spite. 
Sadly, we dont live in a perfect world. A cargo tech cant just... Print a bluespace cell from the autolathe despite their trials to obtain a design disk for them. Bluespace crystals are not allowed in our autolathes, And we're still sitting with snippits of code that displays it _should_ be capable of receiving bluespace crystals but yet, We cannot sate them of their hunger.

Also like its kinda odd it cant. Its like the one material besides biomass you cant thats still *in* basemats datums. Hell, even the ones that are references can be put in autolathes like runite or adamantine, But bluespace crystals? whazzat? Y'cant? Oh well those dont exist for some reason. We got a specific part in a proc for it? Sorry doesn't exist to us. 

image

^ code snippit in question ^

</details>

Basically it allows BS Crystals to be put into the autolathe.

## Why It's Good For The Game

<details>
  <summary>More ramblings</summary>
um. *i dunno*.
Like, Legit making an argument about why is kinda hard because its running against the argument of why not, Which... I dont fully understand. Is there a reason to _not_ allow bluespace crystals into an autolathe? You're already blacklisting any of the "problematic" designs from being ported over to the autolathe that might need them. Is there a side-effect to being in the item material flag? As far as i know there isn't besides being put into material storage components. Does it affect the crystal themselves? well, No it shouldn't. There's a hard-written use factor to over-ride the craft menu so it shouldn't really affect its operation 
But at the same time everything you can make with custom mats doesn't like.... Gain anything from it. You can make a bluespace toolbox to whack someone with but it wont send them to oblivion, You can make magical looking knight armor that does... One of the worst defenses i've seen from plated armor *but you can make it*. Its blue it looks funny all i'm doing is flicking a switch. 

Its probably also not a balance change but like i'm kinda done to death waxing on whether its a balance change somewhere else, so if it isn't technically a balance change just tell me what it is and i'll correct it, first PR up here lmao y'know? Like i just got my git updated and so i was/am expecting technical problems but so far nothings came up so i'm just gonna go with it, y'know?

</details>

TLDR: Why shouldn't BSCrystals be allowed in the autolathe? Every other base material can, And there are some things that do require BSCrystal material that isn't super game breaking. So why not? 

## Changelog

:cl:
balance: Bluespace crystals can now go in ITEM_MATERIAL capable material storage (like autolathes)
/:cl:

### CODE REVIEW

**Comment by VladinXXV** [comment]
>There's a hard-written use factor to over-ride the craft menu so it shouldn't really affect its operation

do we even really need that?
the refined crystals (what you get when removing from the silo) can't be crushed to begin with
using them in hand just tells you that you need to break one off, which is a different behavior.
why do we need that warning to replace the crafting menu? why not both? or just the crafting menu?
it's dumb and i want to mandate bluespace shitters in every RD office

**Comment by Zenitheevee** [comment]
> > There's a hard-written use factor to over-ride the craft menu so it shouldn't really affect its operation
> 
> do we even really need that? the refined crystals (what you get when removing from the silo) can't be crushed to begin with using them in hand just tells you that you need to break one off, which is a different behavior. why do we need that warning to replace the crafting menu? why not both? or just the crafting menu? it's dumb and i want to mandate bluespace shitters in every RD office

Not my own insert but it is what it is.

However, that will not stop me from playing devils advocate, because on just about any standpoint i could understand:

-The fact you can snap it apart with your hands tells me it probably wont have the _best_ structural stability. "Ok but glass can-" There are like 20 different types of glass with different strengths. Lead glass, Crystal, Lead-free crystal, Some can barely withstand weight, Some can support crushing depths of the Marianas trench and the vacuum of space. Considering its the same glass used in making windows that dont immediately break when used as an external structure point, Its *probably* something close to Pyrex (or, well, old pyrex made from Borosilicate, which is actually the same glass used in water-bearing windows that are made of glass and not acrylic).

-It'd be probably *not* the best to break. Its a crystal, and depending on the crystaline structure it could be either naturally sharp (think flint/obsidian) or extremely resistant to being sharp (think: corrundum), But until there's bluespace crystal lore i'm going to probably go with the former because they tend to have weaker crystal strengths.

-Considering bluespace crystal's properties, even if made inert via silo'ing and pulling out, *probably not the best for structures*, especially considering it seems to regain its properties when ground up (See: Bluespace reagent) and as such its probably assumable that its stability comes from being in high enough concentrations.

**Comment by VladinXXV** [comment]
> > > There's a hard-written use factor to over-ride the craft menu so it shouldn't really affect its operation
> > 
> > 
> > do we even really need that? the refined crystals (what you get when removing from the silo) can't be crushed to begin with using them in hand just tells you that you need to break one off, which is a different behavior. why do we need that warning to replace the crafting menu? why not both? or just the crafting menu? it's dumb and i want to mandate bluespace shitters in every RD office
> 
> Not my own insert but it is what it is.
> 
> However, that will not stop me from playing devils advocate, because on just about any standpoint i could understand:
> 
> -The fact you can snap it apart with your hands tells me it probably wont have the _best_ structural stability. "Ok but glass can-" There are like 20 different types of glass with different strengths. Lead glass, Crystal, Lead-free crystal, Some can barely withstand weight, Some can support crushing depths of the Marianas trench and the vacuum of space. Considering its the same glass used in making windows that dont immediately break when used as an external structure point, Its _probably_ something close to Pyrex (or, well, old pyrex made from Borosilicate, which is actually the same glass used in water-bearing windows that are made of glass and not acrylic).
> 
> -It'd be probably _not_ the best to break. Its a crystal, and depending on the crystaline structure it could be either naturally sharp (think flint/obsidian) or extremely resistant to being sharp (think: corrundum), But until there's bluespace crystal lore i'm going to probably go with the former because they tend to have weaker crystal strengths.
> 
> -Considering bluespace crystal's properties, even if made inert via silo'ing and pulling out, _probably not the best for structures_, especially considering it seems to regain its properties when ground up (See: Bluespace reagent) and as such its probably assumable that its stability comes from being in high enough concentrations.

Fair points, however, consider the following:
Walls made of pizza
Toilets made of ground meat
Statues made of plasma, which I wouldn't consider particularly stable either (hence stabilized plasma being an entirely different reagent than plasma)

**Comment by VladinXXV** [comment]
> > There's a hard-written use factor to over-ride the craft menu so it shouldn't really affect its operation
> 
> do we even really need that? the refined crystals (what you get when removing from the silo) can't be crushed to begin with using them in hand just tells you that you need to break one off, which is a different behavior. why do we need that warning to replace the crafting menu? why not both? or just the crafting menu? it's dumb and i want to mandate bluespace shitters in every RD office

also what im talking about here is probably outside the scope of the pr, i just think it'd also be a fitting change, possibly in a followup.

**Comment by Zenitheevee** [comment]
> > > There's a hard-written use factor to over-ride the craft menu so it shouldn't really affect its operation
> > 
> > 
> > do we even really need that? the refined crystals (what you get when removing from the silo) can't be crushed to begin with using them in hand just tells you that you need to break one off, which is a different behavior. why do we need that warning to replace the crafting menu? why not both? or just the crafting menu? it's dumb and i want to mandate bluespace shitters in every RD office
> 
> also what im talking about here is probably outside the scope of the pr, i just think it'd also be a fitting change, possibly in a followup.

Probably, TBH a lot of the material datums look like they need some _**SERIOUS**_ overhauling, But i'm not that deep into coding.
Hell, the only reason i'm acknowledging it as a _**balance**_ PR is because it... By all purposes, Can shift the game balance a bit. Any head can go to the bridge, and if given a design disk (Which, Lets be honest, Isn't hard. Slip science some cash or gun or ammo or tech or w/e have you), Could port a good handful of designs over to an autolathe like cells or even better parts.
And while, Sure, This will usually mostly end up as just cargo or engineering doing it, I dont think many people would deny Head of medical or security their medicinal/recreational autolathe board and basic parts for it.

**Comment by Zenitheevee** [comment]
oh
boy
cant wait to have to Re PR this.
